scipy/spatial/transform/rotation.pyx

```diff
@@ -1357,6 +1460,80 @@ cdef class Rotation(object):
 
         return angles[0] if self._single else angles
 
+    def as_mrp(self):
+        """Represent as Modified Rodrigues Parameters (MRPs).
+
+        MRPs are a 3 element attitude description commonly used in the aerospace
+        industry. It is similar to the rotation vector, except instead of scaling
+        the rotation vector by the angle of rotation, is it scaled by tan(theta/4).
+
+        MRPs have a singuarity at 360 degrees which can be easily avoided by ensuring
+        that the angle of rotation is always <= 180. This function will always return
+        MRPs corresponding to a rotation of <= 180.
+
+        Returns
+        -------
+        mrps : ndarray, shape (3,) or (N, 3)
+            Shape depends on shape of inputs used for initialization.
+
+        References
+        ----------
+        .. [1] Shuster, M. D. "A Survery of Attitude Representations",
+               The Journal of Astronautical Sciences, Vol. 41, No.4, 1993,
+               pp. 475-476
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Represent a single rotation:
+
+        >>> r = R.from_rotvec([0, 0, np.pi])
+        >>> r.as_mrp()
+        array([0.        , 0.        , 1.         ])
+        >>> r.as_mrp().shape
+        (3,)
+
+        Represent a stack with a single rotation:
+
+        >>> r = R.from_euler('xyz', [[180, 0, 0]], degrees=True)
+        >>> r.as_mrp()
+        array([[1.       , 0.        , 0.         ]])
+        >>> r.as_mrp().shape
+        (1, 3)
+
+        Represent multiple rotations:
+
+        >>> r = R.from_rotvec([[np.pi/2, 0, 0], [0, 0, np.pi/2]])
+        >>> r.as_mrp()
+        array([[0.41421356, 0.        , 0.        ],
+               [0.        , 0.        , 0.41421356]])
+        >>> r.as_mrp().shape
+        (2, 3)
+
+        Notes
+        -----
+
+        .. versionadded:: 1.6.0
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double[:, :] mrps = _empty2(num_rotations, 3)
+        cdef int sign
+
+        for ind in range(num_rotations):
+
+            # Ensure we are calculating the set of MRPs that correspond
+            # to a rotation of <= 180
+            sign = -1 if self._quat[ind, 3] < 0 else 1
+
+            for i in range(3):
+                mrps[ind, i] = sign * self._quat[ind, i] / (1 + sign * self._quat[ind, 3])
```

Comment:
Moving `cdef double denominator = 1 + sign * self._quat[ind, 3]` outside of the loop is a reasonable option.